pkg/daemon/discover/discover.go

```diff
@@ -372,9 +403,51 @@ func probeDevices(context *clusterd.Context) ([]sys.LocalDisk, error) {
 		device.Filesystem = fs
 		device.Empty = clusterd.GetDeviceEmpty(device)
 
+		// Add the information provided by ceph-volume inventory
+		if cvInventory != nil {
+			CVData, device_exists := (*cvInventory)[path.Join("/dev/", device.Name)]
+			if device_exists {
+				device.CephVolumeData = CVData
+			} else {
+				logger.Errorf("ceph-volume information for device %s not found", device.Name)
+			}
+		} else {
+			device.CephVolumeData = ""
+		}
+
 		devices = append(devices, *device)
 	}
 
 	logger.Infof("available devices: %+v", devices)
 	return devices, nil
 }
+
+func GetCephVolumeInventory(context *clusterd.Context) (*map[string]string, error) {
+	inventory, err := context.Executor.ExecuteCommandWithOutput(false, "", "ceph-volume", "inventory", "--format", "json")
+	if err != nil {
+		return nil, fmt.Errorf("failed to execute ceph-volume inventory: %+v", err)
+	}
+
+	//Return a map with the information of each device indexed by path
+	CVDevices := make(map[string]string)
+
+	// Get a slice to store the json data
+	bInventory := []byte(inventory)
+	var CVInventory []CephVolumeInventory
+	err = json.Unmarshal(bInventory, &CVInventory)
+	if err != nil {
+		logger.Errorf("error unmarshaling json data coming from ceph-volume inventory: %v", err)
+		return &CVDevices, nil
```

Comment:
`return &CVDevices, fmt.Errorf("error unmarshaling json data coming from ceph-volume inventory. %v", err)`

Reply:
thx!! Much better in this way.